drivers/sensor/grow_r502a/grow_r502a.c

```diff
@@ -461,8 +461,8 @@ static int fps_search(const struct device *dev, uint8_t char_buf_idx)
 		.pid = R502A_COMMAND_PACKET,
 		.data = {R502A_SEARCH, char_buf_idx}
 	};
-	sys_put_be16(R02A_LIBRARY_START_IDX, &tx_packet.data[1]);
-	sys_put_be16(R502A_DEFAULT_CAPACITY, &tx_packet.data[3]);
+	sys_put_be16(R02A_LIBRARY_START_IDX, &tx_packet.data[2]);
+	sys_put_be16(R502A_DEFAULT_CAPACITY, &tx_packet.data[4]);
```

Comment:
`drivers: sensor: grow_r502a: improvements and bug fixes` should be split up into separate commits for each logical bug fix